metric-anomaly-task-manager/src/main/java/org/hypertrace/alert/engine/metric/anomaly/task/manager/job/DbRuleSource.java

```diff
@@ -0,0 +1,161 @@
+package org.hypertrace.alert.engine.metric.anomaly.task.manager.job;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.protobuf.Value;
+import com.google.protobuf.util.JsonFormat;
+import com.typesafe.config.Config;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.tuple.Pair;
+import org.hypertrace.alert.engine.metric.anomaly.datamodel.rule.source.RuleSource;
+import org.hypertrace.config.service.v1.ConfigServiceGrpc;
+import org.hypertrace.config.service.v1.ConfigServiceGrpc.ConfigServiceBlockingStub;
+import org.hypertrace.config.service.v1.ContextSpecificConfig;
+import org.hypertrace.config.service.v1.GetAllConfigsRequest;
+import org.hypertrace.core.documentstore.Document;
+import org.hypertrace.core.documentstore.JSONDocument;
+import org.hypertrace.core.grpcutils.client.RequestContextClientCallCredsProviderFactory;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+import org.hypertrace.core.serviceframework.spi.PlatformServiceLifecycle;
+import org.hypertrace.notification.config.service.v1.GetAllNotificationRulesRequest;
+import org.hypertrace.notification.config.service.v1.NotificationRule;
+import org.hypertrace.notification.config.service.v1.NotificationRuleConfigServiceGrpc;
+
+public class DbRuleSource implements RuleSource {
+
+  private static final String EVENT_CONDITION_DATA_KEY = "eventConditionData";
+  private static final String EVENT_CONDITION_MUTABLE_DATA_KEY = "eventConditionMutableData";
+  private static final String NOTIFICATION_RULE_MUTABLE_DATA_KEY = "notificationRuleMutableData";
+  private static final String METRIC_ANOMALY_DATA_KEY = "metricAnomalyEventCondition";
+  private static final String ID = "id";
+  private static final String CONFIG_SERVICE_HOST_CONFIG_KEY = "config.service.host";
+  private static final String CONFIG_SERVICE_PORT_CONFIG_KEY = "config.service.port";
+  private static final String RESOURCE_NAME_CONFIG_KEY = "resourceName";
+  private static final String RESOURCE_NAMESPACE_CONFIG_KEY = "resourceNamespace";
+  private static final String TENANT_IDS_CONFIG_KEY = "tenantIds";
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  private final ConfigServiceBlockingStub configServiceBlockingStub;
+  private final NotificationRuleConfigServiceGrpc.NotificationRuleConfigServiceBlockingStub
```

Comment:
nit: Both config and notification rule services are using blocking stub, so can we have a consistent name - say `notificationRuleBlockingStub`